src/hii_rail.py

```diff
@@ -127,29 +142,49 @@ def osm_vmap_combined_influence(self):
         )
 
     def vmap_influence(self):
-        vmap_direct_weight = self.rail_weights["vmap_rails"]
-        vmap_indirect_weight = vmap_direct_weight * (
-            self.DIRECT_INDIRECT_INFLUENCE_RATIO
+        vmap_band_names = self.vmap.bandNames()
+        vmap_weights = ee.Dictionary(self.railway_weights["vmap"])
+
+        band_names = ee.Dictionary(
+            {
+                "vmap": vmap_band_names.filter(
+                    ee.Filter.inList("item", vmap_weights.keys())
+                ),
+            }
         )
 
-        self.rail_direct_cost = (
-            self.vmap.distance(kernel=self.kernel["DIRECT"], skipMasked=False)
-            .lte((self.DIRECT_INFLUENCE_WIDTH - self.NOMINAL_ROAD_WIDTH) / 2)
-            .multiply(vmap_direct_weight)
-            .rename("rail_direct")
+        vmap_direct_weights = vmap_weights.toImage().select(band_names.get("vmap"))
+
+        vmap_indirect_weights = vmap_direct_weights.multiply(
+            self.DIRECT_INDIRECT_INFLUENCE_RATIO
+        ).select(band_names.get("vmap"))
+
+        rail_direct_cost = (
+            self.vmap.select(band_names.get("vmap"))
+            .distance(kernel=self.kernel["DIRECT"], skipMasked=False)
+            .lte((self.DIRECT_INFLUENCE_WIDTH - self.NOMINAL_RAIL_WIDTH) / 2)
+            .multiply(vmap_direct_weights)
+            # .reduce(ee.Reducer.max())
```

Comment:
Why are these lines commented? Don't we need these?

Reply:
these can actually be deleted. I had to move the max reducer to a subsequent step.